octoprint_mrbeam/migrate.py

```diff
@@ -214,24 +217,48 @@ def run(self):
                 ):
                     self.hostname_helper_scripts()
 
+                if (
+                    self.beamos_date is not None
+                    and BEAMOS_LEGACY_DATE
+                    < self.beamos_date
+                    <= datetime.strptime("2021-07-19", "%Y-%m-%d").date()
+                    and (self.plugin._settings.get(["version"]) is None)
+                ):  # for images before the 19.7.2021
+                    self.fix_s_series_mount_manager()
+
                 # migrations end
 
-                self.save_current_version()
                 self._logger.info(
                     "Finished migration from v{} to v{}.".format(
                         self.version_previous, self.version_current
                     )
                 )
+                if self.reboot_needed:
+                    self._logger.info("reboot needed, will reboot now")
+                    exec_cmd("sudo reboot now")
```

Comment:
We need to write the version *Before* the reboot

Reply:
ohh yes, fixed that